src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Ok, so its the signature. let's change the signature: IControlUpdate, IButtonUpdate which we can document as having coolodwn in miliseconds. brb doing that.